lib/client/doc.js

```diff
@@ -1036,12 +1036,15 @@ Doc.prototype._hardRollback = function(err) {
 
   // Fetch the latest version from the server to get us back into a working state
   var doc = this;
-  this._fetch({force: true}, function() {
+  this._fetch({force: true}, function(fetchError) {
     // We want to check that no errors are swallowed, so we check that:
     // - there are callbacks to call, and
     // - that every single pending op called a callback
     // If there are no ops queued, or one of them didn't handle the error,
     // then we emit the error.
+
+    fetchError && logger.error('Hard rollback doc fetch failed.', this.version, fetchError);
```

Comment:
I generally find this a bit more readable:

```suggestion
    if (fetchError) logger.error('Hard rollback doc fetch failed.', this.version, fetchError);
```

Also why are we logging `this.version` here and not any other information about the document? (I'd have thought an anonymous document's version on its own is pretty useless information).

I also wonder if we should `emit('error', fetchError)`?